processor/src/test/java/org/mapstruct/ap/test/builder/lifecycle/BuilderLifecycleCallbacksTest.java

```diff
@@ -43,13 +48,15 @@ public void lifecycleMethodsShouldBeInvoked() {
         assertThat( context.getInvokedMethods() )
             .contains(
```

Comment:
@filiphr 
At first I tried to validate the fix by using `containsExactly(...)`. But it soon became too complex and hard to understand due to the nested nature of this test.

So I added an additional fixture test that covers this.

Do you think this should be in a separate `Issue3678Mapper` or is it ok to test it here?

Reply:
I removed the fixture, added the before/afterMapping only with source for the sake of completeness here.

But i provided a separate test that explicitly checks if before and after mapping methods are not called more than once.

Reply:
If we have a dedicated test do we even need changes in this particular test here?

Reply:
Not necessariliy. It was however easier to debug the internals of the method resolution due to multiple lifecycle callback methods. I will remove it after implementing the fix.